internal/compiler/program.go

```diff
@@ -1058,8 +1059,20 @@ func (p *Program) getSemanticDiagnosticsForFileNotFilter(ctx context.Context, so
 		})
 	}
 
+	filtered, directivesByLine := p.getDiagnosticsWithPrecedingDirectives(sourceFile, diags)
+	for _, directive := range directivesByLine {
```

Comment:
I'm slightly confused about why this got moved here instead of in `getDiagnosticsWithPrecedingDirectives`. Would this mean that if a file has a `@ts-expect-error` and the next line has an `includeProcessing` error, then we will error for it here? Though this scenario is pretty convoluted.

Also, would be nice to add a test to make sure this is working and continues to work, since apparently we don't have tests for this from Strada (or maybe we haven't ported them yet).